### ISSUE TITLE
The newline-before-return rule does not fix CRLF files

### ISSUE THREAD

**Issue by tasos-ch** [palantir/tslint#4864]
### Bug Report

- __TSLint version__: 5.20.0
- __TypeScript version__: 3.6.3
- __Running TSLint via__: CLI

#### Reproduction using [TSLint Playground](https://palantir.github.io/tslint-playground)

https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcIBmBXAdmALgSwmgBJjAM6YAUAHgJT7AA6B+2S+lN9j++ATjJitwIUA3Ay4BfMTz4Ch+ANT4AjKLTiQAGnB4k2AOYIQAeiP4AmhBT4wAQwIBbaMwCehABYxrO-QJs48+B68+ADu2Jhu+M6W3PgQIWgAdAwm+ACyNgDWnsQCnhF+UTH4AFbEAdjE+ABuNgA22FAa+HZQTCzRVtzoTGiYECmmEBEwsZW5JM2Y3K5gHmCZ2Gh67p6QaOV1nkgQsaPcu8TJaAwcXFx0IDAURGhQxJfw+ADal5jEDX3wvJD29jB3GBQS4AXQ0UguIG6WweCFoEPOkLQMBCnxgAFoAEYwHa8dG8fiCR6EbowBHifCpGxQNrUqDhXBoeo8FAwlp3VbYWLrXR6Xz+E5qBggcRAA

And run the following command (assuming you have named your file `test.ts`):
```
tslint --fix test.ts
```

#### TypeScript code being linted

```ts
function test(x) {
  if (x) {
    return x;
  }
  return x + 1;
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "newline-before-return": true
  }
}
```

#### Actual behavior

Running --fix leaves the file unmodified.

The console output is the following:
```
Fixed 1 error(s) in index.ts

ERROR: test.ts:5:3 - Missing blank line before return
```

#### Expected behavior

The file should be modified like this:
```typescript
function test(x) {
  if (x) {
    return x;
  }

  return x + 1;
}
```

A blank line should be added before the second `return` statement.

#### Additional notes

After debugging this issue a little bit I have found the following.

**1** The error occurs for CRLF files. LF files are fixed successfully.

**2** I think the issue can be pinpointed to `newLineWithIndentation()` found in `src/utils.ts`.

In particular, when I modified line 319 from:
```typescript
const maybeCarriageReturn =
        sourceFile.text[sourceFile.getLineEndOfPosition(node.pos) - 1] === "\r" ? "\r" : "";
```

to:
```typescript
const maybeCarriageReturn =
        sourceFile.text[sourceFile.getLineEndOfPosition(node.pos)] === "\r" ? "\r" : "";
```

the fix seemed to apply correctly.

However, I would like to note that in no way I am suggesting the above as a fix to this issue (if this proves to be an issue indeed). I am only presenting it to assist in debugging.

Also, it might help you to know the following information about the `newLineWithIndentation()` call:
```
node: NodeObject {
  pos: 18,
  end: 50,
  flags: 0,
  modifierFlagsCache: 0,
  transformFlags: 0,
  parent: NodeObject {
    pos: 16,
    end: 70,
    flags: 0,
    modifierFlagsCache: 0,
    transformFlags: 0,
    parent: NodeObject {
      pos: 0,
      end: 70,
      flags: 0,
      modifierFlagsCache: 536870912,
      transformFlags: 0,
      parent: [SourceFileObject],
      kind: 240,
      asteriskToken: undefined,
      name: [IdentifierObject],
      typeParameters: undefined,
      parameters: [Array],
      body: [Circular]
    },
    kind: 219,
    multiLine: true,
    statements: [ [Circular], [NodeObject], pos: 18, end: 67 ]
  },
  kind: 223,
  expression: IdentifierObject {
    pos: 26,
    end: 27,
    flags: 0,
    modifierFlagsCache: 0,
    transformFlags: 0,
    parent: [Circular],
    escapedText: 'x'
  },
  thenStatement: NodeObject {
    pos: 28,
    end: 50,
    flags: 0,
    modifierFlagsCache: 0,
    transformFlags: 0,
    parent: [Circular],
    kind: 219,
    multiLine: true,
    statements: [ [NodeObject], pos: 30, end: 45 ]
  },
  elseStatement: undefined
}
```
```
sourceFile: SourceFileObject {
  pos: 0,
  end: 70,
  flags: 0,
  modifierFlagsCache: 0,
  transformFlags: 0,
  parent: undefined,
  kind: 285,
  text: 'function test(x) {\r\n  if (x) {\r\n    return x;\r\n  }\r\n  return x + 1;\r\n}',
  bindDiagnostics: [],
  bindSuggestionDiagnostics: undefined,
  languageVersion: 1,
  fileName: 'index.ts',
  languageVariant: 0,
  isDeclarationFile: false,
  scriptKind: 3,
  pragmas: Map {},
  checkJsDirective: undefined,
  referencedFiles: [],
  typeReferenceDirectives: [],
  libReferenceDirectives: [],
  amdDependencies: [],
  hasNoDefaultLib: false,
  statements: [
    NodeObject {
      pos: 0,
      end: 70,
      flags: 0,
      modifierFlagsCache: 536870912,
      transformFlags: 0,
      parent: [Circular],
      kind: 240,
      asteriskToken: undefined,
      name: [IdentifierObject],
      typeParameters: undefined,
      parameters: [Array],
      body: [NodeObject]
    },
    pos: 0,
    end: 70
  ],
  endOfFileToken: TokenObject {
    pos: 70,
    end: 70,
    flags: 0,
    modifierFlagsCache: 0,
    transformFlags: 0,
    parent: [Circular],
    kind: 1
  },
  externalModuleIndicator: undefined,
  nodeCount: 17,
  identifierCount: 5,
  identifiers: Map { 'test' => 'test', 'x' => 'x' },
  parseDiagnostics: [],
  lineMap: [ 0, 20, 32, 47, 52, 69 ]
}
```
```
linesCount: 1
```
```
sourceFile.text[sourceFile.getLineEndOfPosition(node.pos) - 1]: {
```
```
sourceFile.text[sourceFile.getLineEndOfPosition(node.pos) - 1] === "\r": false
````

Thank you in advance for your replies!

Please, let me know if need anything more.

**Comment by JoshuaKGoldberg** [comment]
Hi @tasos-ch, thanks so much for the detailed error information and investigation! 

Per #4534 we're closing issues that just deal with whitespace & formatting related rules. Working with CRLF and LF differences in a Git repository can be a bit of a pain and we don't want to take that on given how close TSLint is to being deprecated.

You're welcome to release a forked custom version of the `newline-before-return` rule if it's useful for other folks.

**Comment by tasos-ch** [comment]
Thank you very much for your reply @JoshuaKGoldberg!

And thank you for the explanation!

Yes, I will make a fork and post the link back here for anyone interested.


> P.S.: Totally agree with the merging of TSLint and ESLint. Best of luck and I hope the merging is completed as soon as possible!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
